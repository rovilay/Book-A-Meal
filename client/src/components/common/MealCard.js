import React, { Component } from 'react';
import PropTypes from 'prop-types';

class MealCard extends Component {
  render() {
    const { mealData, onSubmit } = this.props;
    return (
      <div className="menu-box">
        <img src={mealData.image} alt="menu02" />
        <form className="menu-info" onSubmit={onSubmit}>
          <h3>
            {mealData.title}
          </h3>
          <p className="bold">Price (&#8358;):
            <span> {mealData.price}</span>
          </p>
          <p>{mealData.description}</p>
          <p className="bold">
              Portion: &nbsp;<input type="number" min="1" id="Qty" className="Qty" />
          </p>
          <p>
            <input type="submit" id="order" className="order" value="order" />
          </p>
        </form>
      </div>
    );
  }
}

MealCard.propTypes = {
  mealData: PropTypes.object.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default MealCard;

