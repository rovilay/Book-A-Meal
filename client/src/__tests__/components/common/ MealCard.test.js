import React from 'react';
import { shallow } from 'enzyme';
import { meals } from '../../__mockData__/mealMock';
import MealCard from '../../../components/common/MealCard';


const props = {
  addToCart: jest.fn(),
  editMeal: jest.fn(),
  deleteMeal: jest.fn(),
  cart: meals,
  mealData: meals[0]
};

describe('MealCard component test', () => {
  const setup = () => shallow(<MealCard {...props} />);

  it('should render MealCard correctly', (done) => {
    localStorage.setItem('user', {
      admin: true
    });

    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();

    done();
  });


  it('should call `addToCart` on button click', (done) => {
    localStorage.setItem('user', {
      admin: false
    });

    props.cart = [];
    const wrapper = setup();

    wrapper.find('.responsive-btn-2').first().simulate('click')
    expect(props.addToCart).toHaveBeenCalled();

    done();
  });


  it('should call `handleDeleteMeal`', (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: () => jest.fn()
    };

    const handleDeleteMealSpy= jest.spyOn(wrapper.instance(), 'handleDeleteMeal');
    wrapper.instance().handleDeleteMeal(event);
    expect(handleDeleteMealSpy).toHaveBeenCalled();

    done();
  });
});
