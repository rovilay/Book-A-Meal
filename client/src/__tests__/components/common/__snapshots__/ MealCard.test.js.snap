// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealCard component test should render MealCard correctly 1`] = `
<div
  className="mealCard-container"
>
  <img
    alt="EBA AND EGUSI"
    className="meal-img"
  />
  <div
    className="overlay1"
  >
    <span
      className="meal-title"
    >
      EBA AND EGUSI
    </span>
  </div>
  <div
    className="overlay"
  >
    <div
      className="meal-desc"
    >
      <h4 />
    </div>
  </div>
  <div
    className="meal-label"
  >
    <button
      className="responsive-btn-2"
      disabled={true}
      onClick={[Function]}
    >
      Meal In Cart
    </button>
    <span
      className="meal-price"
    >
      â‚¦ 
      1000
    </span>
  </div>
</div>
`;
